pkg/forman/forman.go

```diff
@@ -0,0 +1,101 @@
+package forman
+
+import (
+	"time"
+
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/db"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/types"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/utils"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/workers"
+
+	"go.uber.org/zap"
+)
+
+type provider struct {
+	working    bool
+	next       *time.Time
+	alarmClock *time.Timer
+}
+
+func GoForman(
+	logger *zap.Logger,
+	client *db.Database,
+	workerCount int,
+	reconcile func(*zap.Logger, *db.Database, types.Request) types.Result,
```

Comment:
Please see prev reviews for reasoning
```suggestion
func GoForman(
	logger *zap.Logger,
	workerCount int,
	reconcile func(*zap.Logger, *db.Database, types.Request) types.Result,
```

Reply:
if we dont pass the db.Database Here, how the foreman can call the below code

```go
logger.Info("Starting", zap.Int("workers", workerCount))
		for i := 0; i < workerCount; i++ {
			workers.GoWorker(toWorkers, results, func(req types.Request) types.Result {
				return reconcile(logger, client, req)
			})
		}
```

Reply:
I am closing this thread as I mentioned how I would like this to look like (with code suggestions) on a newer review